examples/evaluate.py

```diff
@@ -0,0 +1,253 @@
+    )
+    splits: List[str] = wilds_dataset.split_dict.keys()
+    if "train" in splits:
+        splits.remove("train")
+
+    replicates_results: Dict[str, Dict[str, List[float]]] = dict()
+    replicates: List[Union[str, int]] = get_replicates(dataset_name)
+    metrics: List[str] = get_metrics(dataset_name)
+
+    # Store the results for each replicate
+    for split in splits:
+        replicates_results[split] = {}
+        for metric in metrics:
+            replicates_results[split][metric] = []
+
+        for replicate in replicates:
+            predictions_file = get_best_prediction_filename(
+                dataset_name, split, replicate
+            )
+            print(
+                f"Processing split={split}, replicate={replicate}, predictions_file={predictions_file}..."
+            )
+            full_path = os.path.join(path, predictions_file)
+            predicted_labels: List[Any] = get_predictions(full_path)
+            predicted_labels_tensor: torch.Tensor = torch.from_numpy(
+                np.array(predicted_labels)
+            )
+            metric_results: Dict[str, float] = evaluate(
+                wilds_dataset, split, predicted_labels_tensor
+            )
+            for metric in metrics:
+                replicates_results[split][metric].append(metric_results[metric])
+
+    aggregated_results: Dict[str, Dict[str, float]] = dict()
+
+    # Aggregate results of replicates
+    for split in splits:
+        aggregated_results[split] = {}
+        for metric in metrics:
+            replicates_metric_values: List[float] = replicates_results[split][metric]
+            aggregated_results[split][f"{metric}_std"] = np.std(
+                replicates_metric_values, ddof=1
+            )
+            aggregated_results[split][metric] = np.mean(replicates_metric_values)
+
+    # Write out aggregated results to output file
+    print(f"Writing aggregated results for {dataset_name} to {output_path}...")
+    with open(os.path.join(output_path, f"{dataset_name}_results.json"), "w") as f:
+        json.dump(aggregated_results, f, indent=4)
+
+    return aggregated_results
+
+
+def evaluate(
+    dataset: WILDSDataset, split: str, predicted_labels: torch.Tensor
+) -> Dict[str, float]:
+    """
+    Evaluate the given predictions and return the appropriate metrics.
+
+    Parameters:
+        dataset (WILDSDataset): A WILDS Dataset
+        split (str): split we are evaluating on
+        predicted_labels (torch.Tensor): Predictions
+
+    Returns:
+        Metrics as a dictionary with metrics as the keys and metric values as the values
+    """
+    # Dataset will only be downloaded if it does not exist
+    subset: WILDSSubset = dataset.get_subset(split)
+    true_labels: torch.Tensor = subset.y_array
+    metadata: torch.Tensor = subset.metadata_array
+    # Attempt to resize predicted_labels tensor to match true_labels tensor's shape
+    predicted_labels.resize_(true_labels.shape)
```

Comment:
Sorry, why is this necessary? If the predictions are of the wrong shape, we should probably flag it?

Reply:
When I read the predictions from the csv files, I stored it as an array, so I needed to expand the dimensions before comparing. This should the same as using Torch's `unsqueeze`.

Reply:
Ah got it. I have a slight preference for explicitly unsqueezing if that's ok, since it's easier to understand why we're doing it by reading the code (and if the shapes are unexpectedly wrong, then it won't just automatically go through because of the resize).